scripts/check_compatibility.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+python_versions=("3.8" "3.9" "3.10" "3.11")
+all_passed=true
+
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+
+check_compatibility() {
+    python_version=$1
+    all_supported=0
+
+    while read -r requirement; do
+        package_name=$(echo "$requirement" | awk -F'[!=<>]' '{print $1}')
+        echo -n "Checking $package_name... "
+
+        url="https://pypi.org/pypi/$package_name/json"
+        response=$(curl -s $url)
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" $url)
+
+        if [ "$status_code" != "200" ]; then
+            echo -e "${RED}Information not available${NC}"
+            continue
+        fi
```

Comment:
should this be treated like a non-compatible scenario?
if we cannot ensure this automatically we have to do something at least (like allow listing package versions compatibility even if its harcoded). But maybe we do not want to have green builds and red logs, right?

Becuase if all is OK, but one package is not found, we will have a green build with this error-red message. But we will not see it most probably.

What do you think about that?